src/main/java/jenkins/metrics/util/AutoSamplingHistogram.java

```diff
@@ -59,11 +67,23 @@ public AutoSamplingHistogram(Gauge<? extends Number> source, Reservoir reservoir
     }
 
     public void update() {
-        Number value = source.getValue();
-        if (value instanceof Integer) {
-            update(value.intValue());
-        } else {
-            update(value.longValue());
+        try {
+            Number value = source.getValue();
+            if (value instanceof Integer) {
+                update(value.intValue());
+            } else if (value != null) {
+                update(value.longValue());
+            } else {
+                LOGGER.log(Level.FINE, "Gauge {0} returned null", source);
+            }
+            badGauge = false;
+        } catch (ClassCastException e) {
+            LogRecord lr = new LogRecord(badGauge ? Level.FINE : Level.WARNING,
+                    "Gauge {0} is supposed to return a subclass of java.lang.Number but didn't");
+            badGauge = true;
```

Comment:
The `badGauge` getting set after the `LogRecord` is created + the `badGauge ? Level.FINE : Level.WARNING` seems a bit quirky.